src/test/util/setup_common.cpp

```diff
@@ -139,10 +139,11 @@ BasicTestingSetup::BasicTestingSetup(const ChainType chainType, TestOpts opts)
     // data directories use a random name that doesn't overlap with other tests.
     SeedRandomForTest(SeedRand::FIXED_SEED);
 
+    const std::string test_name{G_TEST_GET_FULL_NAME ? G_TEST_GET_FULL_NAME() : ""};
     if (!m_node.args->IsArgSet("-testdatadir")) {
-        // By default, the data directory has a random name
-        const auto rand_str{g_rng_temp_path.rand256().ToString()};
-        m_path_root = fs::temp_directory_path() / TEST_DIR_PATH_ELEMENT / rand_str;
+        // By default, the data directory has a random name on each test run
+        const auto rand_str{util::ToString(GetTime<std::chrono::nanoseconds>().count())};
```

Comment:
non-blocking nit:
Since time is being used now, the name isn't quite as random as before. Could update the comment and `rand_str` name.

Reply:
sure, pushed. Thanks